File: 1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complement = {}
        for i in range(len(nums)):
            if target - nums[i] in complement:
                return [complement[target - nums[i]], i]
            complement[nums[i]] = i

File: 1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        vector<int> res;
        for (auto i = 0; i < nums.size(); ++i) {
            auto found = hash.find(target - nums[i]);
            if (found != hash.end()) {
                res.push_back(found->second);
                res.push_back(i);
            }          
            hash.insert({nums[i], i});
        }
        return res;
    }
};
